error_formatter.go

```diff
@@ -36,8 +37,10 @@ type ErrorFormatter interface {
 	SetColorFormatter(color ColorFormatter)
 }
 
+// ColorFormatter represents a function that writes to the terminal using color.
 type ColorFormatter func(w io.Writer, f string, a ...interface{}) (n int, err error)
 
+// TODO: Explain why this is needed.
 var _ ErrorFormatter = &termErrorFormatter{}
```

Comment:
I was tempted to change this to `var _ = &termErrorFormatter{}`. The type seems redundant.

Reply:
This is a Go idiom for ensuring that a struct implements an interface. 
More info: https://golang.org/doc/effective_go.html#blank_implements